.github/workflows/preprod-deployment.yml

```diff
@@ -83,20 +85,41 @@ jobs:
           SHORT_SHA: ${{ steps.vars.outputs.sha }}
 
       - name: Upload the VCS Agent binary to downloads.spacelift.dev
+        if: ${{ github.ref == 'refs/heads/main' }}
         run: >-
           aws s3 sync
           ${BIN_DIR} s3://${{ secrets.PREPROD_AWS_S3_BUCKET }}/
           --no-progress
 
       - name: Invalidate downloads.spacelift.dev cache
+        if: ${{ github.ref == 'refs/heads/main' }}
         run: >-
           aws cloudfront create-invalidation
           --distribution-id ${{ secrets.PREPROD_DISTRIBUTION }}
           --paths "/*"
 
+      - name: Log in to Amazon public ECR
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
+    #  - name: Set up QEMU
```

Comment:
Maybe add a comment here, explaining this will be needed for adding multi-arch build support in future?